src/route-handlers/list-workflows/list-workflows.ts

```diff
@@ -0,0 +1,56 @@
+import { type NextRequest, NextResponse } from 'next/server';
+
+import decodeUrlParams from '@/utils/decode-url-params';
+import * as grpcClient from '@/utils/grpc/grpc-client';
+import type { Props, RouteParams } from './list-workflows.types';
+import listWorkflowsQueryParamSchema from './schemas/list-workflows-query-params-schema';
+import getListWorkflowExecutionsQuery from './helpers/get-list-workflow-executions-query';
+import mapExecutionsToWorkflows from './helpers/map-executions-to-workflows';
+
+export async function listWorkflows(request: NextRequest, props: Props) {
```

Comment:
props and Props types can be changed to a better name to separate them from components props.